samtranslator/model/api/api_generator.py

```diff
@@ -805,8 +805,7 @@ def _create_basepath_mapping(
         basepath_mapping.DomainName = ref(api_domain_name)
         basepath_mapping.RestApiId = ref(rest_api.logical_id)
         basepath_mapping.Stage = ref(rest_api.logical_id + ".Stage")
-        if basepath:
-            basepath_mapping.BasePath = basepath
+        basepath_mapping.BasePath = basepath
```

Comment:
I think we still need this if, because we use this method passing `None` as `basepath` (and `basepath` is optional), so we should probably actually update this to `if basepath is not None:` instead of removing the `if`. 

I think that's a different edge case that we might not be covering in the test cases either, which is the first branch of this `if`: https://github.com/aws/serverless-application-model/blob/develop/samtranslator/model/api/api_generator.py#L546-L548